include/aos/common/tools/memory.hpp

```diff
@@ -44,25 +47,40 @@ class SmartPtr {
 
         assert(!object || allocator);
```

Comment:
dont understand the need for this assert

Reply:
Valid cases:

object && allocator
!object && allocator
!object && !allocator

This is short condition to cover them all.